ionic2/src/components/type/type.component.ts

```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from '@angular/core';
+
+import { TypeService } from '../../services/type.service';
+
+@Component({
+  selector: 'poke-type',
+  templateUrl: './type.component.html'
+})
+export class TypeComponent {
+  @Input() type: string;
+  constructor(private typeService: TypeService) { }
```

Comment:
!For future reference only, i'll fix it asap.
I missed something there: This should be public.
Reason being: If you use the service in a template, the AoT compile must be able to access it.